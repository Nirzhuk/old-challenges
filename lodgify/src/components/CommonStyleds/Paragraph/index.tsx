import styled from "styled-components";

interface ParagraphProps {
    size?: string;
    color?: string;
    as?: string;
}

const Paragraph = styled.p<ParagraphProps>`
  font-size: ${({ size }) => size};
  color: ${({ color }) => color};
`;

Paragraph.defaultProps = {
    size: "initial",
    color: "black",
};

export default Paragraph;
