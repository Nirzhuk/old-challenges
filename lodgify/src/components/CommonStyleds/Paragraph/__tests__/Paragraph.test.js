import { render, screen } from '@testing-library/react'
import Paragraph from '..';


test('Paragraph normal', async () => {
    const { container } = render(<Paragraph>booked</Paragraph>)
    expect(container.firstChild).toMatchSnapshot()
    expect(screen.getByText("booked")).toHaveStyle(`color:black`)

})

test('Paragraph green', async () => {
    const { container } = render(<Paragraph color="#11A960">booked</Paragraph>)
    expect(container.firstChild).toMatchSnapshot()
    expect(screen.getByText("booked")).toHaveStyle(`color:#11A960`)
})
test('Paragraph big size', async () => {
    const { container } = render(<Paragraph size="24px" color="#11A960">booked</Paragraph>)
    expect(container.firstChild).toMatchSnapshot()
    expect(screen.getByText("booked")).toHaveStyle(`color:#11A960`)
    expect(screen.getByText("booked")).toHaveStyle(`font-size:24px`)
})