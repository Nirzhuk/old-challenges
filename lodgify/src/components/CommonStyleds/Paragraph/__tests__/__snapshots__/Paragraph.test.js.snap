// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph big size 1`] = `
<p
  class="sc-bdnxRM hsJSvg"
  color="#11A960"
>
  booked
</p>
`;

exports[`Paragraph green 1`] = `
<p
  class="sc-bdnxRM bQocks"
  color="#11A960"
>
  booked
</p>
`;

exports[`Paragraph normal 1`] = `
<p
  class="sc-bdnxRM gpuXDS"
  color="black"
>
  booked
</p>
`;
